src/game/server/props.cpp

```diff
@@ -1050,7 +1050,7 @@ int CBreakableProp::OnTakeDamage( const CTakeDamageInfo &inputInfo )
 		return 1;
 	}
 
-	if( info.GetAttacker() && info.GetAttacker()->MyCombatCharacterPointer() )
+	if ( !IsOnFire() && ToBaseCombatCharacter( info.GetAttacker() ) != NULL )
 	{
 		m_hLastAttacker.Set( info.GetAttacker() );
```

Comment:
Doesn't this already have the intended behavior of setting the last attacker whenever a prop takes damage? Now, if the prop is on fire and get insta-detonated by another player, the credit goes to the person who set the prop on fire